### PR TITLE
Make phpunit fail on deprecation warnings

### PR DESCRIPTION
**Pull request by staabm**

### CODE REVIEW

**Comment by staabm** [comment]
interessting.. locally I get phpunit fail instead of accepting the error. I think this means the deprecation warning is not emitted at all in CI

**Comment by staabm** [comment]
ok, we can see CI failling now. will re-revert the initial fix now

**Comment by staabm** [comment]
@nikic I think this fix should cover us in the future

**Review by samsonasik — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -49,6 +49,7 @@ jobs:
         with:
           coverage: "none"
           php-version: "${{ matrix.php-version }}"
+          ini-values: "error_reporting=2147483647" # E_ALL
```
Comment (nikic) [inline .github/workflows/main.yml:52]:
```suggestion
          ini-values: "error_reporting=E_ALL"
```
Does this not work?

**Comment by staabm** [comment]
opened a setup-php bug https://github.com/shivammathur/setup-php/issues/827

File: phpunit.xml.dist [hunk 2]
```diff
@@ -4,6 +4,10 @@
          xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
          backupGlobals="false"
          colors="true"
+         convertErrorsToExceptions="true"
+         convertNoticesToExceptions="true"
+         convertWarningsToExceptions="true"
+         convertDeprecationsToExceptions="true"
```
Comment (nikic) [inline phpunit.xml.dist:10]: I think you only need this one, the rest default to true.

File: .github/workflows/main.yml (hunk 1 above)
Comment (staabm) [reply .github/workflows/main.yml]: seems to work

Comment (staabm) [reply .github/workflows/main.yml]: using the fix [suggested by the setup-php author](https://github.com/shivammathur/setup-php/issues/827#issuecomment-2002367665) now

**Comment by nikic** [comment]
Quite unexpected behavior to me ... guess it's time to go through all my repos and add `ini-file: "development"` to CI actions.

**Comment by staabm** [comment]
> Quite unexpected behavior to me

agree. the author agrees to. its planned for the next major, because its a breaking change.

see https://github.com/shivammathur/setup-php/issues/827#issuecomment-2002367665

**Pull request merged**